src/main/java/org/jabref/gui/backup/BackupResolverDialog.java

```diff
@@ -0,0 +1,37 @@
+package org.jabref.gui.backup;
+
+import java.nio.file.Path;
+
+import javafx.scene.control.ButtonBar;
+import javafx.scene.control.ButtonType;
+import javafx.scene.layout.Region;
+
+import org.jabref.gui.FXDialog;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.BackupFileType;
+import org.jabref.logic.util.io.BackupFileUtil;
+
+public class BackupResolverDialog extends FXDialog {
+    public static final ButtonType RESTORE_FROM_BACKUP = new ButtonType(Localization.lang("Restore from backup"), ButtonBar.ButtonData.OK_DONE);
+    public static final ButtonType REVIEW_BACKUP = new ButtonType(Localization.lang("Review backup"), ButtonBar.ButtonData.LEFT);
+    public static final ButtonType IGNORE_BACKUP = new ButtonType(Localization.lang("Ignore backup"), ButtonBar.ButtonData.CANCEL_CLOSE);
+
+    public BackupResolverDialog(Path originalPath) {
+        super(AlertType.CONFIRMATION, Localization.lang("Backup found"), true);
+        setHeaderText(null);
+        getDialogPane().setMinHeight(Region.USE_PREF_SIZE);
+
+        Path backupPath = BackupFileUtil.getPathOfLatestExisingBackupFile(originalPath, BackupFileType.BACKUP).orElseThrow();
+
+        // TODO: Localize this string once the localization consistency tests are updated to support Java 15 text blocks
+        setContentText("""
+                        A backup file for '%s' was found at '%s'.
+                        This could indicate that JabRef did not shut down cleanly last time the file was used.
+
+                        Do you want to recover the library from the backup file?
+                        """.formatted(originalPath.getFileName().toString(),
```

Comment:
Trying to localize text blocks led to a flailing test with the exception below. I don't completely understand why it's failing. It seems that when scanning java files for localized strings, the localization parser counts the prefix spaces of a text block line as natural spaces.

If that's the case then there is nothing I can do other than add a TODO until it's resolved.
```java
LocalizationConsistencyTest > findMissingLocalizationKeys() FAILED
    java.lang.RuntimeException: "
                            A backup file for '%0' was found at '%1'.
                            This could indicate that JabRef did not shut down cleanly last time the file was used.

                            Do you want to recover the library from the backup file?
                            " ends with a space. As this is a localization key, this is illegal!
        at org.jabref.logic.l10n.JavaLocalizationEntryParser.getLanguageKeysInString(JavaLocalizationEntryParser.java:38)
        at org.jabref.logic.l10n.LocalizationParser.getLanguageKeysInJavaFile(LocalizationParser.java:156)
        at org.jabref.logic.l10n.LocalizationParser.lambda$findLocalizationEntriesInJavaFiles$2(LocalizationParser.java:94)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:273)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at org.jabref.logic.l10n.LocalizationParser.findLocalizationEntriesInJavaFiles(LocalizationParser.java:95)
        at org.jabref.logic.l10n.LocalizationParser.findLocalizationEntriesInFiles(LocalizationParser.java:71)
        at org.jabref.logic.l10n.LocalizationParser.findMissingKeys(LocalizationParser.java:34)
        at org.jabref.logic.l10n.LocalizationConsistencyTest.findMissingLocalizationKeys(LocalizationConsistencyTest.java:123)
```

Reply:
Have you tried with \n for inserting newlines instead of using textblocks? That should work

Reply:
From a quick look seems like we need to adjust our regexes...  Could be a nice issue for students, doing TDD etc. 
https://github.com/JabRef/jabref/blob/2497ba4d82b08c207ee083ede3fdfeac307bb0e2/src/test/java/org/jabref/logic/l10n/JavaLocalizationEntryParser.java#L12

Reply:
Note that we support single-line strings only. Was a decision of 2016 to keep the parser simple.

We tried to switch to full Java parsing at https://github.com/JabRef/jabref/pull/8050/files#diff-3c2319710ad0e37f5f3b8d774be9847eea6ac95bba4faaef1b5bfe47188aaf6c (by using [JavaParser](https://github.com/javaparser/javaparser)). However, we had issues with new Java syntax back then (`yield` statement, .... This was solved two weeks ago.) Not sure whether we rely often on new Java features which cannot be parsed by JavaParser.

Reply:
@HoussemNasri Could you rewrite that as single line string? 😇 Linebreaks are IMHO encoded using `\n` in the localization file.

Reply:
Well, in 2016 I think there were textblocks not yet part of java language...
As Java lang is currently heavily evolving i agree it's probably best to stay for now with our currently simple parser and eventually move to a third party dependency solution as the language syntax stabilizes again in one or two years...